executor/apply_cache.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/hashicorp/golang-lru"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/memory"
+	"github.com/pingcap/tidb/util/stringutil"
+)
+
+type applyCache struct {
+	cache       *lru.Cache
+	memCapacity int64
+	memTracker  *memory.Tracker // track memory usage.
+}
+
+type applyCacheValue struct {
+	Data *chunk.List
+}
+
+var applyCacheLabel fmt.Stringer = stringutil.StringerStr("applyCache")
+
+func newApplyCache(ctx sessionctx.Context) (*applyCache, error) {
+	num := int(ctx.GetSessionVars().ApplyCacheCapacity / 100)
```

Comment:
What does `num` mean?

The count of the cached element or the size of the total memory usage of applyCache?

Reply:
`num` means the count of the cached element. I will add some comments to this.